src/services/refactors/convertReactPureComponent.ts

```diff
@@ -0,0 +1,294 @@
+            StatelessComponent: undefined,
+        } as any;
+        let importClause: ImportClause;
+        sourcefile.forEachChild(c => {
+            if (!isImportDeclaration(c)) { return; }
+            if (!c.importClause) { return; }
+            if ((c.moduleSpecifier as StringLiteral).text !== "react") { return; }
+            const { name, namedBindings } = c.importClause;
+            importClause = c.importClause;
+            const names: (keyof typeof result)[] = ["Component", "PureComponent", "SFC", "StatelessComponent"];
+            // In case of:
+            // import * as ns from "mod" => name = undefined, namedBinding: NamespaceImport = { name: ns }
+            if (namedBindings && isNamespaceImport(namedBindings)) { result.React = namedBindings.name.text; }
+            // import d from "mod" => name = d, namedBinding = undefined
+            else if (name) { result.React = name.text; }
+            // import d, * as ns from "mod" => name = d, namedBinding: NamespaceImport = { name: ns }
+            // ? No this case
+            // import { a, b as x } from "mod" => name = undefined, namedBinding: NamedImports = { elements: [{ name: a }, { name: x, propertyName: b}]}
+            if (namedBindings && isNamedImports(namedBindings)) {
+                namedBindings.elements.forEach(e => {
+                    if (e.propertyName) {
+                        const p: keyof typeof result = e.propertyName.text as any;
+                        if (names.indexOf(p) !== -1) { result[p] = e.name.text; }
+                    }
+                    else {
+                        const n: keyof typeof result = e.name.text as any;
+                        if (names.indexOf(n) !== -1) { result[n] = n; }
+                    }
+                });
+            }
+            // import d, { a, b as x } from "mod" => name = d, namedBinding: NamedImports = { elements: [{ name: a }, { name: x, propertyName: b}]}
+            // ? No this case
+        });
+        if (result.React) {
+            result.Component = result.Component || result.React + ".Component";
+            result.PureComponent = result.PureComponent || result.React + ".PureComponent";
+            result.SFC = result.SFC || result.React + ".SFC";
+            result.StatelessComponent = result.StatelessComponent || result.StatelessComponent + ".StatelessComponent";
+        }
+        else if (!result.Component && !result.PureComponent && !result.React && !result.SFC && !result.StatelessComponent) {
+            return undefined;
+        }
+        return { ...result, importClause: importClause! };
+    }
+
+    function isReactSFCDeclaration(node: Node, sourcefile: SourceFile): Component | undefined {
+        if (!isSFCLikeDeclaration(node)) { return; }
+        if (node.asteriskToken) { return; }
+        if (!node.body) { return; }
+
+        // TODO: Should also check the actual type insteadof only receive explicit typed
+        const typeNode = node.type;

```

Comment:
we should not be relying on type annoation. tsserver support JS experience as well. so these apply to both JS and TS files.

we should isntead try to "guess" if this is an SFC or not..
- funciton declaration
  - or const declaration with initializer function expression or lambda
- Zero, one or two parameters
- function returnns a JSX element (this is gonna be a bit tricky, but we can start by saying the expression of all return statements must be a JSX element).

Reply:
We can go further and say it has to be at the top-level of the file.

Reply:
Yes, in todo I'll try to guess if it is an SFC after I done the explicit typed SFC. 

`context`, `propTypes`, `contextTypes`, `defaultProps` and `displayName` are also in later stage of pr.

Reply:
And, hmm, what does it mean by `at the top-level of the file`? Like it is one of the SourceFile's children?